Source/JavaScriptCore/jit/ExecutableAllocator.cpp

```diff
@@ -1387,10 +1387,10 @@ void dumpJITMemory(const void* dst, const void* src, size_t size)
     uint64_t dst64 = std::bit_cast<uintptr_t>(dst);
     uint64_t size64 = size;
     TraceScope(DumpJITMemoryStart, DumpJITMemoryStop, time, dst64, size64);
-    DumpJIT::write(&time, sizeof(time));
-    DumpJIT::write(&dst64, sizeof(dst64));
-    DumpJIT::write(&size64, sizeof(size64));
-    DumpJIT::write(src, size);
+    DumpJIT::write(asByteSpan(time));
+    DumpJIT::write(asByteSpan(dst64));
+    DumpJIT::write(asByteSpan(size64));
+    DumpJIT::write(unsafeMakeSpan(static_cast<const uint8_t*>(src), size));
```

Comment:
Please use `sourceSpan` to avoid the unverified length here.

Reply:
Hi @geoffreygaren, you didn't expand the source code enough.  :)

`sourceSpan` is used in the `DumpJIT::write()` method, which is being called here.  The entire `struct DumpJIT` implementation is declared inside this method.

The `src` and `size` in this call come from the `dumpJITMemory()` function arguments on [line 1320](https://github.com/WebKit/WebKit/pull/52678/files#diff-47b7950aaa32998f5ef88ee18bc31dec5970a6f655ca79ec8e02f2ad4570d0a2R1320):
```
void dumpJITMemory(const void* dst, const void* src, size_t size)
```